Taken wholesale from NetBSD: patch-deps_v8_src_base_atomicops__internals__arm__gcc.h
--- node-v4.2.2.orig/deps/v8/src/base/atomicops_internals_arm_gcc.h     2015-11-03 20:00:07.000000000 +0000
+++ node-v4.2.2/deps/v8/src/base/atomicops_internals_arm_gcc.h  2015-11-27 13:02:32.846550000 +0000
@@ -13,6 +13,11 @@
 #include <sys/cpuinline.h>
 #endif
 
+#if defined(__FreeBSD__)
+#include <sys/types.h>
+#include <machine/sysarch.h>
+#endif
+
 namespace v8 {
 namespace base {
 
@@ -50,6 +55,8 @@
   ((KernelMemoryBarrierFunc)0xffff0fa0)();
 #elif defined(__QNXNTO__)
   __cpu_membarrier();
+#elif defined(__FreeBSD__)
+  sysarch(ARM_DRAIN_WRITEBUF, (void *)0);
 #else
 #error MemoryBarrier() is not implemented on this platform.
 #endif
